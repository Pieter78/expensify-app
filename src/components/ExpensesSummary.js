

// 1. Create ExpensesSummary component

// Rendered by ExpensesDashboardPage
// Test with two snapshot tests

// Connected to store for:
import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = props => (
  <div>
    <p>test</p>
    {
      `You are viewing ${props.expenses.length} items for a total of ${numeral(props.total / 100).format('$0,0.00')}`
    }
  </div>
);

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters),
  total: expensesTotal(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpensesSummary);


// ExpensesCount
// ExpensesTotal

// Example
// Viewing 2 expenses, totalling $94.34
