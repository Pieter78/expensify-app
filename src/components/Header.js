import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';

export const Header = ({ clickLogout }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink activeClassName="is-active" exact to="/">Dashboard</NavLink>
    <NavLink activeClassName="is-active" exact to="/create">Create expense</NavLink>
    <button onClick={clickLogout}>Logout</button>
  </header>
);

const mapDispatchToProps = dispatch => ({
  clickLogout: () => dispatch(startLogout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
