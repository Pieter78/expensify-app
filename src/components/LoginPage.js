import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ clickLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It&#39;s time to bring your expenses under control!</p>
      <button className="button button--blue" onClick={clickLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  clickLogin: () => {
    dispatch(startLogin());
  },
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
