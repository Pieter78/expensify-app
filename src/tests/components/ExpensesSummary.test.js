import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render correctly with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={199} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render correctly with several expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={10} expensesTotal={122299} />);
  expect(wrapper).toMatchSnapshot();
});
