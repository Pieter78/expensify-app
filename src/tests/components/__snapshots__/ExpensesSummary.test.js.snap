// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       for a total of 
      <span>
        $1.99
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button button--blue"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render correctly with several expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        10
      </span>
       
      expenses
       for a total of 
      <span>
        $1,222.99
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button button--blue"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
